apps/src/applab/ImportProjectDialog.jsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+
+import Dialog, {Body, Buttons, Confirm, Footer} from '../templates/Dialog';
+import color from '../color';
+import {fetchProject} from './redux/screens';
+
+const styles = {
+  urlInputWrapper: {
+    display: 'flex',
+    alignItems: 'stretch',
+    width: '100%',
+  },
+  urlInput: {
+    width: 'inherit',
+  },
+  // TODO: ditch these styles in favor of standardized typography components
+  // once they exist
+  instructions: {
+    color: color.black,
+  },
+  errorText: {
+    color: color.red,
+  },
+};
+
+const ImportProjectDialog = React.createClass({
+
+  propTypes: Object.assign({}, Dialog.propTypes, {
+    onImport: React.PropTypes.func.isRequired,
+    isFetching: React.PropTypes.bool,
+    error: React.PropTypes.bool,
+  }),
+
+  getDefaultProps() {
+    return {
+      isFetching: false,
+      error: false,
```

Comment:
well, I think in theory if they are not passed in, they will be null. I suppose the way they are being used makes that perfectly fine.